src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -286,6 +292,23 @@ module.exports = function<T, P, I, TI, C, CX>(config : HostConfig<T, P, I, TI, C
   function commitAllLifeCycles() {
     while (nextEffect) {
       const current = nextEffect.alternate;
+
+      // Renderers may schedule work to be done after host components are mounted
+      // (eg DOM renderer may schedule auto-focus for inputs and form controls).
+      // These effects should only be committed when components are first mounted,
+      // aka when there is no current/alternate.
+      if (
+        !current &&
+        nextEffect.tag === HostComponent &&
```

Comment:
Hmm how about we move this to `commitLifeCycles`? Avoids an extra check. https://github.com/facebook/react/blob/master/src/renderers/shared/fiber/ReactFiberCommitWork.js#L434